core/tx_pool.go

```diff
@@ -558,7 +558,8 @@ func (pool *TxPool) validateTx(tx *types.Transaction, local bool) error {
 		return ErrInvalidGasPrice
 	}
 	// Heuristic limit, reject transactions over 32KB to prevent DOS attacks
-	if tx.Size() > 32*1024 {
+	// UPDATED to 64KB to support the deployment of bigger contract due to the pressing need for sophisticated/complex contract in financial/capital markets - Nathan Aw
+	if tx.Size() > 64*1024 {
```

Comment:
IMHO, it would be best to have this value be passed in as a cli parameter. @nathanawmk mind updating?

Reply:
It could be added to the genesis configuration, since all nodes would need to agree what the max size of a contract is. If it's different then some nodes may accept whilst others reject the transaction.

Reply:
Agreed; we should set this via genesis configuration

Reply:
Do we also need to change the use of `MaxCodeSize` while we're at it? It seems like this check in tx_pool might only be a quick liberal check closer to the networking side only to be later rejected via `MaxCodeSize` in the EVM?

See:
https://github.com/jpmorganchase/quorum/blob/87d7c906e9eba93b618651ab5a427746506e2f30/params/protocol_params.go#L56
https://github.com/jpmorganchase/quorum/blob/0af7ad0664e1d5d7367eaa47e00dbe0501a526dd/core/vm/evm.go#L416

Reply:
I was trying MaxCodeSize in the genesis.json but it did not work thus I had to update the tx.size. What is for sure we need to update the parameters in order for complex and sophisticated smart contract logic to be deployed. E.g., Trade Reconciliation. Please let me know how I can take this forward so others won't have to go through what I had. Thank you!

Reply:
@bts evm size check is for the data returned from running an operation. @nathanawmk @bts @prd-fox  Updating the genesis block would only take effect for new networks, not existing ones.

Reply:
Thanks! Any possible downstream impact if i modify the tx.size? Anyway can this pull request be accepted?

Reply:
> IMHO, it would be best to have this value be passed in as a cli parameter. @nathanawmk mind updating?

Hi @fixanoid thanks. How should one best approach passing the max tx size in a CLI parameter?

Reply:
> Thanks! Any possible downstream impact if i modify the tx.size? Anyway can this pull request be accepted?

Please complete the CLA and we'll merge it. The size limit is to mostly protect against DOS attacks on mainnet.